skills/studio/scripts/studio/utils/error_codes.py

```
@@ -116,4 +116,18 @@
 # Content language validation
 # ---------------------------------------------------------------------------
 CONTENT_LANGUAGE_VIOLATION = "LANG001"
+
+# ---------------------------------------------------------------------------
+# CDSL structure — CDSL.md FAIL rules (S.3-7, CL.1-4, CO.4-6)
+# ---------------------------------------------------------------------------
+CDSL_MISSING_CHECKBOX = "cdsl-missing-checkbox"
+CDSL_MISSING_PHASE_TOKEN = "cdsl-missing-phase-token"
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to: "CDSL_MISSING_PHASE_TOKEN"

(S105)